.github/workflows/test.yml

```diff
@@ -0,0 +1,310 @@
+name: SalesforceCommerceCloud/pwa-kit/test
+on:
+  schedule:
+    - cron: 0 8 * * *
+  push:
+     branches:
+     # TODO: Before merging remove branches filter
+     - github-actions
+jobs:
+  pwa-kit:
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [14]
+        npm: [6, 7, 8]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node }}
+          cache: npm
+
+      - name: Save NPM version
+        run: |-
+            echo "NPM_VERSION=${{ matrix.npm }}" >> $GITHUB_ENV
+        shell: bash
+
+      - name: Setup Ubuntu Machine
+        uses: "./.github/actions/setup_ubuntu"
+
+      - name: Install node dependencies with NPM version
+        if: matrix.npm != 6
+        run: |- 
+          npm install -g npm@${{ matrix.npm }}
+          npm ci
+
+      - name: NPM version
+        run: npm --version
+
+      - name: Run tests
+        uses: "./.github/actions/runtests"
+
+      - name: Run Lighthouse CI on the PWA
+        if: matrix.npm == 6
+        uses: "./.github/actions/lighthouse-ci"
+
+      - name: Smoke test scripts
+        if: matrix.npm == 6
+        uses: "./.github/actions/smoketestscripts"
+
+      - name: Create MRT credentials file
```

Comment:
It seems like we extracted some steps into their own actions, for example `./.github/actions/runtests`, `./.github/actions/smoketestscripts`, could we do it to all steps? This `test.yml` is a bit hard to read given its length.